pkg/storage/ingest/reader.go

```diff
@@ -169,11 +170,28 @@ func (r *PartitionReader) consumeFetches(ctx context.Context, fetches kgo.Fetche
 		})
```

Comment:
Question: is it always guaranteed to get at least 1 record in `fetches` or should we check it and skip the function if there are no records?

Reply:
not after https://github.com/grafana/mimir/pull/6951; changed